libensemble/libE.py

```diff
@@ -100,6 +100,46 @@
             y:
                 type: float
 
+On macOS and Windows, when using multiprocessing (``local`` comms), you may need to place most
```

Comment:
I would not say this is about leaked semaphore warnings or pytorch. Rather, without it, is isn't going to work. 

I would say more like the warning line for tests - Something along the lines of - If using local comms and the process start method is 'Spawn' (used by Windows, and the default on macOS since python 3.8), then you must use `if __name__ == "__main__:". You could add the that bracketed part - also the startmethod required by pytorch, perhaps.

Reply:
If you want, could also point out the alternative of setting start method to 'fork' on macOS.